shuriken/include/shuriken/api/C/shuriken_core.h

```diff
@@ -111,19 +111,93 @@ SHURIKENCOREAPI void analyze_classes(hDexContext context);
 
 
 /// @brief Obtain one hdvmclassanalysis_t given its hdvmclass_t
-/// @param context context from the CORE API
+/// @param context DEX context from the CORE API
 /// @param class_ hdvmclass_t to get its analysis
 SHURIKENCOREAPI hdvmclassanalysis_t *get_analyzed_class_by_hdvmclass(hDexContext context, hdvmclass_t *class_);
 
 /// @brief Obtain one hdvmclassanalysis_t given its name.
-/// @param context context from the CORE API
+/// @param context DEX context from the CORE API
 /// @param class_name name of the class to retrieve
 SHURIKENCOREAPI hdvmclassanalysis_t *get_analyzed_class(hDexContext context, const char *class_name);
 
+/// @brief Obtain one hdvmmethodanalysis_t given its hdvmmethod_t
+/// @param context DEX context from the CORE API
+/// @param method hdvmmethod_t to get its analysis
 SHURIKENCOREAPI hdvmmethodanalysis_t *get_analyzed_method_by_hdvmmethod(hDexContext context, hdvmmethod_t *method);
 
+/// @brief Obtain one hdvmmethodanalysis_t given its name
+/// @param context DEX context from the CORE API
+/// @param method_full_name dalvik name of the method
 SHURIKENCOREAPI hdvmmethodanalysis_t *get_analyzed_method(hDexContext context, const char *method_full_name);
-;
+
+/// C - APK part of the CORE API from ShurikenLib
+
+///--------------------------- Parser API ---------------------------
+
+/// @brief main method from the APK Core API
+/// it parses the APK file and it retrieves a context object
+/// @param filePath path to the APK file to analyze
+/// @param create_xref `1` to create xrefs, `0` to avoid creating xrefs
+/// @return context object to obtain information from the APK file
+SHURIKENCOREAPI hApkContext parse_apk(const char *filePath, int create_xref);
```

Comment:
instead of an INT for create_xref, an enum would be better fit for the job?

Reply:
Could be, we could create an `enum` for it, since C doesn't have any boolean type by default, I didn't add anything like that, also from the Python API an `int` is provided for this.